src/hidpp20.c

```diff
@@ -2458,6 +2458,17 @@ hidpp20_onboard_profiles_parse(struct hidpp20_device *device,
 							    profiles,
 							    i,
 							    check_crc);
+
+		/* on fail to read the user profile fallback to the firmware profile */
+		if(rc && profiles->profiles[i].address == i + 1)
+		{
+			profiles->profiles[i].address += 0x0100;
+			rc = hidpp20_onboard_profiles_parse_profile(device,
+							    profiles,
+							    i,
+							    check_crc);
```

Comment:
Don't  you need to call check_crc with false here? On the mice I have, the ROM profiles do not have a valid CRC, so this call would fail.

Reply:
TBH I didn't really understand what you were doing in the loop above. `hidpp20_onboard_profiles_read_sector` is only called one time and we are reading multiple profiles?? Doesn't each profile have its own sector?

Reply:
The loop before the reading of the profiles is to read the dictionary of the device, where the addresses of the profiles are stored and where we know if a profile is enabled or not.

Reply:
Yeah nvm, you are just reading the profile directory, that's fine. What do you think about disabling the profile here and in the initial caller trying to enable it again with rom values?

Reply:
We always wanted to have a non intrusive approach. So if the device works, we should not tamper with it. Especially if there is a bug like the other one you just solved where libratbag consider a profile invalid when it is not and you are resetting the mouse for users.

TL;DR: please do not edit the dictionary there

Reply:
I was thinking about disabling the profile inside libratbag but as you point out the internal state in libratbag is tied to the state in the hardware. What do you propose then? How can we see if the profile as a valid CRC in the initial caller?

Reply:
basically unfold the calls to `hidpp20_onboard_profiles_parse()` in `hidpp20_onboard_profiles_initialize()`, and from there change the whole logic.

Reply:
That part I understand but how are you thinking about changing the whole logic? I have been trying to reformulate the logic in my head and I don't see how that would be substantially different from what I am doing here.

Reply:
I am not saying we should rethink the whole commands. The logic is fine.

But in the light of the new logic, the code should be reflecting that new logic only:
For each profile, we try to fetch the in-flash profile, if it fails, try to read the ROM one.

Then done.